framework/src/transfers/MultiAppCopyTransfer.C

```diff
@@ -51,204 +52,99 @@ MultiAppCopyTransfer::initialSetup()
 }
 
 void
-MultiAppCopyTransfer::execute()
+MultiAppCopyTransfer::transferDofObject()
 {
-  _console << "Beginning CopyTransfer " << name() << std::endl;
-
-  switch (_direction)
-  {
-    case TO_MULTIAPP:
+  if (_to_object->n_dofs(_to_sys_num, _to_var_num) > 0) // If this variable has dofs at this node
```

Comment:
That was my original intent, except it needs about 8 different things. So, I didn't want a list of bunch of parameters, for some reason I don't like that.  Also, if this is running on each node and element, I figured it would be more efficient by updating member variables rather than having pass by reference, but I actually don't know if that is true. I will leave that for you CS types, I doesn't make that much of a difference for me.